demos/unity_states/unity/Assets/Scripts/BallStateDataSourcePlatform.cs

```diff
@@ -0,0 +1,224 @@
+{
+    private BallStateDataSourceUnity dataSourceUnity;
+
+    // Platform interop types
+    private TP.Baseline baselineMain;
+    private BallContract ballContract;
+    private TP.Object.ObjectHandle ballObject;
+    private BallStates ballStates;
+
+    private States.StateHandle_F32 stateHandlePosX;
+    private States.StateHandle_F32 stateHandlePosY;
+    private States.StateHandle_F32 stateHandlePosZ;
+    private States.StateHandle_I16 stateHandleEulerX;
+    private States.StateHandle_I16 stateHandleEulerY;
+    private States.StateHandle_I16 stateHandleEulerZ;
+    private States.StateHandle_F32 stateHandleScaleX;
+    private States.StateHandle_F32 stateHandleScaleY;
+    private States.StateHandle_F32 stateHandleScaleZ;
+    private States.StateHandle_U64 stateHandleColor;
+
+    void Awake()
+    {
+        this.dataSourceUnity = GetComponent<BallStateDataSourceUnity>();
+        InstantiatePlatformObject();
+    }
+
+    void Update()
+    {
+        ApplyUnityStateToPlatformBaseline();
+        LogCurrentData();
+    }
+
+    private void InstantiatePlatformObject()
+    {
+        this.baselineMain = new TP.Baseline(true);
+        this.ballContract = BallContract.Register(this.baselineMain);
+        this.ballObject = this.ballContract.ObjectCreate(
+            this.baselineMain,
+            this.dataSourceUnity.pos_x, this.dataSourceUnity.pos_y, this.dataSourceUnity.pos_z,
+            this.dataSourceUnity.euler_x, this.dataSourceUnity.euler_y, this.dataSourceUnity.euler_z,
+            this.dataSourceUnity.scale_x, this.dataSourceUnity.scale_y, this.dataSourceUnity.scale_z,
+            this.dataSourceUnity.color
+        );
+
+        ConfigurePlatformObjectStates();
+    }
+
+    private void ConfigurePlatformObjectStates()
+    {
+        this.ballStates = this.ballContract.States;
+        this.stateHandlePosX = this.baselineMain.BindStateF32(this.ballStates.PosX, this.ballObject);
+        this.stateHandlePosY = this.baselineMain.BindStateF32(this.ballStates.PosY, this.ballObject);
+        this.stateHandlePosZ = this.baselineMain.BindStateF32(this.ballStates.PosZ, this.ballObject);
+        this.stateHandleEulerX = this.baselineMain.BindStateI16(this.ballStates.EulerX, this.ballObject);
+        this.stateHandleEulerY = this.baselineMain.BindStateI16(this.ballStates.EulerY, this.ballObject);
+        this.stateHandleEulerZ = this.baselineMain.BindStateI16(this.ballStates.EulerZ, this.ballObject);
+        this.stateHandleScaleX = this.baselineMain.BindStateF32(this.ballStates.ScaleX, this.ballObject);
+        this.stateHandleScaleY = this.baselineMain.BindStateF32(this.ballStates.ScaleY, this.ballObject);
+        this.stateHandleScaleZ = this.baselineMain.BindStateF32(this.ballStates.ScaleZ, this.ballObject);
+        this.stateHandleColor = this.baselineMain.BindStateU64(this.ballStates.Color, this.ballObject);
+    }
+
+    private void ApplyUnityStateToPlatformBaseline()
```

Comment:
A "Data Source" implies that data flows into unity from this source. However, this function actually takes data out of unity and into the component.

I think it makes more sense to turn this component into a component just for the platform, which can be read and written to, with maybe some helper functions. It should not hold any references to other components, it should stand on its own. It implements the data source interface.

Then there is another component for the physics sim. It implements the data source interface.

Whats missing (and currently squished into the current implementation of BallStateDataSourcePlatform) is a third component that synchronizes the two data sources. This lets the other two components not care about each other at all, and all of the middleware logic that connects them is in the new third component.

Reply:
Will separate the Platform data source logic into 2 different classes.